scripts/run.sh

```diff
@@ -22,6 +22,18 @@ function cleanup {
 # NATS
 ###
 gnatsd &
+counter=1
+until nc -zv localhost 4222 1>/dev/null 2>&1; 
+do
+    sleep 0.5
+    ((counter++))
+    if [ ${counter} -gt 20 ]
+    then
+        echo "gnatsd failed, exiting"
+        exit 1
+    fi
+    echo "Waiting for gnatsd"       
```

Comment:
In order not to repeat and reprint the same line over and over until gnats is up and running, I would suggest that you change this line to:
```bash
echo -ne "Waiting for gnatsd"
```
This will practically write the same message over and over on the same line, so that when the script is run it will show:
```
...
Waiting for gnatsd
...
```
instead of:
```
...
Waiting for gnatsd
Waiting for gnatsd
Waiting for gnatsd
Waiting for gnatsd
...
```

Of course this is my subjective view and I think that one message is enough. If we want to show how long does it take for `natsd` to start, than showing multiple messages would be ok as well.
@mteodor feel free to proceed the way you think it would be better.